crates/crabe_decision/src/strategy/testing/test_vision_moveto.rs

```diff
@@ -0,0 +1,79 @@
+use std::ops::Div;
+use nalgebra::{distance, Point2};
+use crabe_framework::data::output::Command;
+use crabe_framework::data::tool::ToolData;
+use crabe_framework::data::world::World;
+use crate::action::ActionWrapper;
+use crate::action::move_to::MoveTo;
+use crate::strategy::Strategy;
+
+const DIST_TARGET_REACHED: f64 = 0.1;
+
+#[derive(Debug)]
+enum TestVisionMoveToStatus {
+    Placement,
+    MovingForward,
+    MovingBackwards,
+}
+
+pub struct TestVisionMoveTo {
+    ids: Vec<u8>,
+    status: TestVisionMoveToStatus,
+    positive_half: bool,
+}
+
+impl TestVisionMoveTo {
+    pub fn new(ids: Vec<u8>, positive_half: bool) -> Self {
+        Self {
+            ids,
+            status: TestVisionMoveToStatus::Placement,
+            positive_half,
+        }
+    }
+
+
+}
+
+impl Strategy for TestVisionMoveTo {
+    fn name(&self) -> &'static str {
+        "TestVisionMoveTo"
+    }
+
+    fn step(&mut self, world: &World, _: &mut ToolData, action_wrapper: &mut ActionWrapper) -> bool {
+        // WARNING : Not clearing the action_wrapper leads to stuttering
+        action_wrapper.clear_all();
+        let sign = if self.positive_half { 1. } else { -1. };
+        let mut y_target = 0.;
+        let mut next_status = TestVisionMoveToStatus::MovingForward;
+        match self.status {
+            TestVisionMoveToStatus::Placement => {
+                y_target = 0.;
+                next_status = TestVisionMoveToStatus::MovingForward;
+            }
+            TestVisionMoveToStatus::MovingForward => {
+                y_target = 1.;
+                next_status = TestVisionMoveToStatus::MovingBackwards;
+            }
+            TestVisionMoveToStatus::MovingBackwards => {
+                y_target = -1.;
+                next_status = TestVisionMoveToStatus::MovingForward;
+            }
+        }
+
+        // Move robots
+        let mut change_status = true;
+
+        world.allies_bot.iter()
```

Comment:
It can be improved by iterating over all ids instead of allies_bots.

Reply:
true, I'll rewrite this ASAP

Reply:
wait no we can't do that. Say I have robot 1 off the field. If I use
```Rust
self.ids.iter()
```
to send commands to robot 1, will it receive the commands even though it isn't on the field ?

Reply:
Looking at `crates/crabe_decision/src/action.rs` we have the following when computing the commands to add to the command_map :
https://github.com/NAMeC-SSL/CRAbE/blob/dbeac2ba43e41255552d97eb061e983ebd1bcd48/crates/crabe_decision/src/action.rs#L92-L95
There's no additional check to whether the robot is on the field (or so it seems).

Maybe we should add a filter at this location so that it does not send any command to off-field robots ? At the same time, this is the kind of caveat we should probably note down somewhere, and even notify the dev if a command was sent to an off-field robot. At 2:00 am in the middle of the night, I might forget that kind of specificity.

Reply:
Great find on a feature that our project lacks. I believe it's the job of crabe_guard to handle this - to remove and notify by a warning or an error.

Simultaneously, we should investigate what happens if a robot doesn't appear within a certain timeframe. I recall that we addressed this in the filter, but I don't remember whether it remains in the world.